crates/bevy_geometry/src/lib.rs

```diff
@@ -0,0 +1,383 @@
+    ///      |  (4)------(0)
+    ///      |   |    |   |
+    ///     (7)--|---(3)  |
+    ///        \ |      \ |
+    ///         (6)------(2)
+    /// ```
+    pub fn vertices(&self) -> [Vec3; 8] {
+        let mut vertices = [Vec3::ZERO; 8];
+        let aab_vertices = self.aab.vertices();
+        for i in 0..vertices.len() {
+            vertices[i] = self.transform.project_point3(aab_vertices[i])
+        }
+        vertices
+    }
+
+    /// Set the oriented box's aab.
+    pub fn set_aabb(&mut self, aab: AABB) {
+        self.aab = aab;
+    }
+
+    /// Set the oriented box's transform.
+    pub fn set_transform(&mut self, transform: Mat4) {
+        self.transform = transform;
+    }
+    pub fn fast_aabb(&self) -> AABB {
+        let vertices = self.vertices();
+        let mut max = Vec3::splat(f32::MIN);
+        let mut min = Vec3::splat(f32::MAX);
+        for vertex in vertices.iter() {
+            max = vertex.max(max);
+            min = vertex.min(min);
+        }
+        // Unwrap is okay here because min < max
+        AABB::from_min_max(min, max).unwrap()
+    }
+}
+
+/// An axis aligned box is a box whose axes lie in the x/y/z directions of the coordinate system
+/// the box is defined in.
+#[derive(Copy, Clone, PartialEq, Debug, Reflect)]
+pub struct AABB {
+    min: Vec3,
+    max: Vec3,
+}
+impl Primitive3d for AABB {
+    fn outside_plane(&self, plane: Plane) -> bool {
+        for vertex in self.vertices().iter() {
+            if plane.distance_to_point(*vertex) <= 0.0 {
+                return false;
+            }
+        }
+        true
+    }
+}
+impl AABB {
+    /// An ordered list of the vertices that form the 8 corners of the [AxisAlignedBox].
+    /// ```none
+    ///          (5)------(1)               Y
+    ///           | \      | \              |
+    ///           |  (4)------(0) MAX       o---X
+    ///           |   |    |   |             \
+    ///      MIN (7)--|---(3)  |              Z
+    ///             \ |      \ |
+    ///              (6)------(2)
+    /// ```
+    pub fn vertices(&self) -> [Vec3; 8] {
+        let min = self.min;
+        let max = self.max;
+        [
+            Vec3::new(max.x, max.y, max.z),
+            Vec3::new(max.x, max.y, min.z),
+            Vec3::new(max.x, min.y, max.z),
+            Vec3::new(max.x, min.y, min.z),
+            Vec3::new(min.x, max.y, max.z),
+            Vec3::new(min.x, max.y, min.z),
+            Vec3::new(min.x, min.y, max.z),
+            Vec3::new(min.x, min.y, min.z),
+        ]
+    }
+    /// Construct an [AxisAlignedBox] given the coordinates of the minimum and maximum corners.
+    pub fn from_min_max(min: Vec3, max: Vec3) -> Result<AABB, PrimitiveError> {
+        if (max - min).min_element() >= 0.0 {
+            Ok(AABB { min, max })
+        } else {
+            Err(PrimitiveError::MinGreaterThanMax)
+        }
+    }
+    /// Construct an [AxisAlignedBox] from the origin at the minimum corner, and the extents - the
```

Comment:
```suggestion
    /// Construct an [`AxisAlignedBox`] from the origin at the minimum corner, and the extents - the
```
having the ticks (`) in the doc comments helps highlight the fact this is a type :)